handlers/smshandler/smshandler.go

```diff
@@ -0,0 +1,274 @@
+package smshandler
+
+import (
+	"encoding/binary"
+	"encoding/csv"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/nyaruka/phonenumbers"
+	"github.com/vocdoni/blind-csp/types"
+	"go.vocdoni.io/dvote/db"
+	"go.vocdoni.io/dvote/db/metadb"
+	"go.vocdoni.io/dvote/log"
+)
+
+// DefaultMaxSMSattempts defines the default maximum number of SMS allowed attempts.
+const DefaultMaxSMSattempts = 5
+
+// SmsHandler is a handler that requires a simple math operation to be resolved.
+type SmsHandler struct {
+	stg                 Storage
+	forceElectionsMatch bool
+	challengeDB         db.Database
+	challengeLock       sync.RWMutex
```

Comment:
why are we guarding all databases with mutexes? I thought our database interfaces were already meant to be safe for concurrent use.

Reply:
this is legacy code, should be removed